### PR TITLE
Constify tup_version

### PR DESCRIPTION
**Pull request by FreddieChopin**
There's no point in this object being a variable.

### CODE REVIEW

**Comment by FreddieChopin** [comment]
As the patch -in my opinion - is trivial, I skipped the CLA thing... If you really need it, I will reconsider.

File: src/tup/version/Tupfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 include_rules
 ifneq (@(TUP_PLATFORM),win32)
-: |> version=`git describe` ; (echo '#include "tup/version.h"'; echo "const char *tup_version = \"$version\";") > %o |> version.c
+: |> version=`git describe` ; (echo '#include "tup/version.h"'; echo "const char * const tup_version = \"$version\";") > %o |> version.c
```
Comment (gittup) [inline src/tup/version/Tupfile]: How about just 'const char tup_version[] = "..."' ?

Comment (FreddieChopin) [inline src/tup/version/Tupfile]: done (;

**Comment by gittup** [comment]
Merged - thanks!

File: src/tup/version.h [hunk 2]
```diff
@@ -22,6 +22,6 @@
 #define tup_version_h
 
 /* This constant is defined by an automatically generated file */
-extern const char *tup_version;
```
Comment (anatol) [inline src/tup/version.h]: I believe you could make the value constant by using 

`extern const char *const tup_version;`

Comment (FreddieChopin) [inline src/tup/version.h]: This was exactly what I did in the first version, but Mike preferred the variant with array. The effect is identical anyway (;

**Pull request closed without merging**